### PR TITLE
check object instance fixes #311

### LINKED ISSUE #311: Fix PHP Warning in EED_Events_Archive_Caff
following error appears in error logs:

> PHP Warning: First parameter must either be an object or the name of an existing class in /caffeinated/modules/events_archive_caff/EED_Events_Archive_Caff.module.php on line 169

plz see: https://events.codebasehq.com/projects/event-espresso/tickets/10139
for further information

### PR DESCRIPTION
**Pull request by tn3rb**
plz see #311
and https://events.codebasehq.com/projects/event-espresso/tickets/10139 for further information

### CODE REVIEW

File: caffeinated/modules/events_archive_caff/EED_Events_Archive_Caff.module.php [hunk 1]
```diff
@@ -1,4 +1,7 @@
-<?php if ( ! defined('EVENT_ESPRESSO_VERSION')) exit('No direct script access allowed');
+<?php use EventEspresso\core\exceptions\InvalidDataTypeException;
```
Comment (nerrad) [inline caffeinated/modules/events_archive_caff/EED_Events_Archive_Caff.module.php]: maybe move this import statement to its own line?

**Comment by tn3rb** [comment]
###testing

 * log into admin and go to **EE** > **Events** > **Templates** tab, scroll down to **Event List Pages**  section
 * edit any settings that pertain to the event archives and save the page
 * navigate away to another page/tab, then return
 * [ ] confirm that setting changes persisted as expected
 * visits event archives on frontend
 * [ ] confirm that everything is as expected

**Comment by tn3rb** [comment]
heh... ok... will check that out

File: caffeinated/modules/events_archive_caff/EED_Events_Archive_Caff.module.php [hunk 2]
```diff
@@ -156,6 +156,9 @@ public static function template_settings_form()
     public static function update_template_settings($CFG, $REQ)
     {
         $config = new EE_Events_Archive_Config();
+        $CFG->EED_Events_Archive = $CFG->EED_Events_Archive instanceof EE_Events_Archive_Config
+            ? $CFG->EED_Events_Archive
+            : $config;
```
Comment (nerrad) [inline caffeinated/modules/events_archive_caff/EED_Events_Archive_Caff.module.php]: Is it possible that there might be two instances of `EE_Events_Archive_Config` floating about?  Reason I ask is `$config` instantiates a new instance of `EE_Events_Archive_Config`, but that is only set on `$CFG->EED_Events_Archive` if there's already no instance set.

Would it be better to maybe do something like this?

```php
$CFG->EED_Events_Archive = $CFG->EED_Events_Archive instanceof EE_Events_Archive_Config
     ? $CFG->EED_Events_Archive
     : new EE_Events_Archive_Config();
$config = $CFG->EED_Events_Archive;
```

Comment (tn3rb) [reply caffeinated/modules/events_archive_caff/EED_Events_Archive_Caff.module.php]: > Is it possible that there might be two instances of EE_Events_Archive_Config floating about?

ya maybe. all of the config stuff is horrible old old legacy code so anything is possible.
I'll clean it up a bit more using your idea

**Comment by tn3rb** [comment]
kk ready for testing again Seth

**Pull request merged**